### PR TITLE
feat(config): workaround for web3j v5.0.0

### LINKED ISSUE #13857: Renovate suggests updating to higher sorted versions regardless of release timestamps
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Renovate suggest updating the dependency `org.web3j:core` from `4.8.9` to `5.0.0`, see https://github.com/cesarsotovalero/besu/pull/39.

However, according to the Maven Central repository, version `5.0.0` was released in 2020, whereas version `4.8.9` is from 2021 (more recent), see https://mvnrepository.com/artifact/org.web3j/core.

Therefore, there is a mismatch between the version and the release date here. 

**Suggested action:** Renovate should detect and alert developers about this caveat. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

### PR DESCRIPTION
**Pull request by cesarsotovalero**
## Changes:

Adds a workaround for web3j v5.0.0

## Context:

Closes #13857

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [X] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/workarounds.ts [hunk 1]
```diff
@@ -35,6 +36,16 @@ export const presets: Record<string, Preset> = {
       },
     ],
   },
+  ignoreWeb3jCoreWithOldReleaseTimestamp: {
+    description: 'Ignore web3j 5.0.0 release',
+    packageRules: [
+      {
+        matchDatasources: ['maven'],
+        matchPackageNames: ['org.web3j:core'],
+        allowedVersions: '/^(?!5\\.0\\.0)/',
```
Comment (viceice) [inline lib/config/presets/internal/workarounds.ts:45]:
```suggestion
        allowedVersions: '!/^5\\.0\\.0/',
```
Renovate uses `!` to invert a Regex.

Comment (cesarsotovalero) [reply lib/config/presets/internal/workarounds.ts:45]: I see. Thanks for this suggestion!

**Comment by rarkins** [comment]
Shouldn't ignoring just 5.0.0 be enough? Eg if 5.1.0 or 6.0.0 are ever released, those would presumably be ok?

**Comment by viceice** [comment]
> Shouldn't ignoring just 5.0.0 be enough? Eg if 5.1.0 or 6.0.0 are ever released, those would presumably be ok?

Isn't the Regex doing this? 🤔

**Comment by cesarsotovalero** [comment]
> Shouldn't ignoring just 5.0.0 be enough? 

Yes, it's just ignoring 5.0.0.

**Comment by rarkins** [comment]
Whoops, you're right - somehow I thought I saw a `>=` when reading on mobile

**Pull request merged**